File: JSBASIC/178_palindrome_arr.js
// 178	178_palindrome_arr.js	Check whether a given array reads the same forwards and backwards.

let arr = [1, 2, 3, 2, 1];
let isPalindrome = true;
for (let i = 0; i < Math.floor(arr.length / 2); i++) {
    if (arr[i] !== arr[arr.length - 1 - i]) { isPalindrome = false; break; }
}
console.log("Array:", arr);
console.log("Is palindrome:", isPalindrome);

File: JavaBasic/178_palindrome_arr.java
// 178	178_palindrome_arr.java	Check whether a given array reads the same forwards and backwards.
import java.util.Arrays;
public class _178_palindrome_arr {
    public static void main(String[] args) {
        int[] arr = {1,2,3,2,1};
        boolean isPalin = true;
        for (int i=0;i<arr.length/2;i++) if(arr[i]!=arr[arr.length-1-i]){isPalin=false;break;}
        System.out.println("Array: " + Arrays.toString(arr));
        System.out.println("Is palindrome: " + isPalin);
    }
}
